src/scoring/hpf_scoring.py

```diff
@@ -0,0 +1,168 @@
+"""The HPF Model scoring class."""
+
+import numpy as np
+import os
+from edward.models import Poisson
+from edward.models import Gamma
+import tensorflow as tf
+from collections import defaultdict
+from src.config import (SCORING_THRESHOLD,
+                        HPF_SCORING_REGION,
+                        HPF_output_package_id_dict,
+                        HPF_output_manifest_id_dict,
+                        HPF_output_rating_matrix,
+                        HPF_output_item_matrix,
+                        a, a_c, c, c_c,
+                        b_c, d_c, K)
+
+
+class HPFScoring:
+    """The HPF Model scoring class."""
+
+    def __init__(self, datastore=None, scoring_threshold=SCORING_THRESHOLD,
+                 scoring_region=HPF_SCORING_REGION):
+        """Set the variables and load model data."""
+        self.datastore = datastore
+        self.scoring_threshold = scoring_threshold
+        self.scoring_region = scoring_region
+        self.package_id_dict = None
+        self.id_package_dict = None
+        self.rating_matrix = None
+        self.beta = None
+        self.manifest_id_dict = None
+        self.manifests = 0
+        self.packages = 0
+        self.rating_mean = 0.0
+        self.a = tf.constant(a)
```

Comment:
You don't really need to store these constants, the call to tf.constant puts them in the computation graph and you can retrieve them from the graph wherever required. Storing would anyway create a copy of the operation, unless that's what you want here.